chore(ecmascript): add missing `unsafe` block to inline assembly call

camc314 opened the pull request:
Fixes the following warning when building oxlint

```
warning[E0133]: use of inline assembly is unsafe and requires unsafe block
  --> crates/oxc_ecmascript/src/to_int_32.rs:47:5
   |
47 | /     std::arch::asm!(
48 | |         "fjcvtzs {dst:w}, {src:d}",
49 | |         src = in(vreg) number,
50 | |         dst = out(reg) ret,
51 | |     );
   | |_____^ use of inline assembly
   |
   = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html>
   = note: inline assembly is entirely unchecked and can cause undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
  --> crates/oxc_ecmascript/src/to_int_32.rs:41:1
   |
41 | unsafe fn f64_to_int32_arm64(number: f64) -> i32 {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: requested on the command line with `-W unsafe-op-in-unsafe-fn`
```

camc314 commented:
* **#12871** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12871?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12871?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12871).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Aug 7, 9:13 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
chore(ecmascript): add missing `unsafe` block to inline assembly call (#12871)

Fixes the following warning when building oxlint

```
warning[E0133]: use of inline assembly is unsafe and requires unsafe block
  --> crates/oxc_ecmascript/src/to_int_32.rs:47:5
   |
47 | /     std::arch::asm!(
48 | |         "fjcvtzs {dst:w}, {src:d}",
49 | |         src = in(vreg) number,
50 | |         dst = out(reg) ret,
51 | |     );
   | |_____^ use of inline assembly
   |
   = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html>
   = note: inline assembly is entirely unchecked and can cause undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
  --> crates/oxc_ecmascript/src/to_int_32.rs:41:1
   |
41 | unsafe fn f64_to_int32_arm64(number: f64) -> i32 {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: requested on the command line with `-W unsafe-op-in-unsafe-fn`
```
